local_units/test_views.py

```diff
@@ -409,21 +423,182 @@ def test_create_update_local_unit_health(self):
+        self.assertEqual(response_updated_1.data["modified_by"], local_unit_obj.created_by.id)
+
+    def test_revert_local_unit(self):
+        region = Region.objects.create(name=2)
+        country = Country.objects.create(name="Philippines", iso3="PHL", iso="PH", region=region)
+        type = LocalUnitType.objects.create(code=2, name="Code 0")
+        level = LocalUnitLevel.objects.create(level=1, name="Code 1")
+        affiliation = Affiliation.objects.create(code=1, name="Code 1")
+        functionality = Functionality.objects.create(code=1, name="Code 1")
+        health_facility_type = FacilityType.objects.create(code=1, name="Code 1")
+        primary_health_care_center = PrimaryHCC.objects.create(code=1, name="Code 1")
+
+        data = {
+            "local_branch_name": "Silele Red Cross Clinic, Sigombeni Red Cross Clinic & Mahwalala Red Cross Clinic",
+            "english_branch_name": None,
+            "type": type.id,
+            "country": country.id,
+            "created_at": "2024-05-13T06:53:14.978083Z",
+            "modified_at": "2024-05-13T06:53:14.978099Z",
+            "draft": False,
+            "validated": True,
+            "postcode": "",
+            "address_loc": "Silele Clinic is is in Hosea Inkhundla under the Shiselweni, Sigombeni is in Nkom'iyahlaba Inkhundla under the Manzini region and Mahwalala is in the Mbabane West Inkhundla under the Hhohho region.",  # noqa: E501
+            "address_en": "",
+            "city_loc": "",
+            "city_en": "",
+            "link": "",
+            "location": "SRID=4326;POINT (31.567837 -27.226852)",
+            "source_loc": "",
+            "source_en": "",
+            "subtype": "",
+            "date_of_data": "2024-05-13",
+            "level": level.id,
+            "location_json": {
+                "lat": 42.066667,
+                "lng": 19.983333,
+            },
+            "health": {
+                "other_affiliation": None,
+                "focal_point_email": "jele@redcross.org.sz",
+                "focal_point_phone_number": "26876088546",
+                "focal_point_position": "Programmes Manager",
+                "other_facility_type": None,
+                "residents_doctor": 0,
+                "feedback": "first question of initial question did not provide for the option to write the name of the NS. It is written LRC yet it should allow Baphalali Eswatini Red Cross Society (BERCS) to be inscribed in the box.",  # noqa: E501
+                "affiliation": affiliation.id,
+                "functionality": functionality.id,
+                "health_facility_type": health_facility_type.id,
+                "primary_health_care_center": primary_health_care_center.id,
+            },
+            "visibility_display": "RCRC Movement",
+            "focal_person_loc": "Elliot Jele",
+            "focal_person_en": "",
+            "email": "",
+            "phone": "",
+        }
```

Comment:
Creating LocalUnitFactory() will be better approach here since we have created local unit in other test functions too.

Reply:
Need to check if the snapshot is created or not for the localunit.